.ci/releaser/src/index.js

```diff
@@ -0,0 +1,314 @@
+      body
+    };
+  });
+
+  return commits;
+}
+
+function analyzeCommits(commits) {
+  return commits.map(commit => {
+    const prefix = commit.subject.split(":", 1)[0].toLowerCase();
+    let bumpType = BUMP_TYPE_PER_PREFIX[prefix];
+
+    if (!bumpType) {
+      bumpType = BUMP_TYPE.NONE;
+    }
+
+    const release = commit.body.includes(RELEASE_MARKER);
+
+    return {
+      commit,
+      bumpType,
+      release
+    };
+  });
+}
+
+function getBumpType(analyzedCommits) {
+  return Math.max(...analyzedCommits.map(el => el.bumpType));
+}
+
+function bumpVersion(version, bumpType) {
+  const parts = version.split(".");
+
+  if (bumpType === BUMP_TYPE.MAJOR) {
+    parts[0]++;
+  } else if (bumpType === BUMP_TYPE.MINOR) {
+    parts[1]++;
+  } else if (bumpType === BUMP_TYPE.PATCH) {
+    parts[2]++;
+  }
+
+  return parts.join(".");
+}
+
+function generateChangelog(analyzedCommits) {
+  const breakingChanges = [];
+  const featureChanges = [];
+  const fixChanges = [];
+
+  for (const commit of analyzedCommits) {
+    let changes;
+    if (commit.bumpType === BUMP_TYPE.MAJOR) {
+      changes = breakingChanges;
+    } else if (commit.bumpType === BUMP_TYPE.MINOR) {
+      changes = featureChanges;
+    } else if (commit.bumpType === BUMP_TYPE.PATCH) {
+      changes = fixChanges;
+    }
+
+    changes.push(commit.commit);
+  }
+
+  const changeTypes = [
+    ["BREAKING CHANGES", breakingChanges],
+    ["Features", featureChanges],
+    ["Fixes", fixChanges]
+  ];
+
+  const changelog = [];
+  for (const type of changeTypes) {
+    const title = type[0];
+    const changes = type[1];
+
+    if (changes.length === 0) {
+      continue;
+    }
+
+    changelog.push("## " + title);
+
+    const innerChangelog = [];
+    for (const change of changes) {
+      innerChangelog.push(`* ${change.subject} (${change.sha})`);
+    }
+    changelog.push(innerChangelog.join("\n"));
+  }
+
+  return changelog.join("\n\n");
+}
+
+async function build(version, sourceCommitSha) {
+  const sourceConvention = await fs.readFile(
+    path.join(ROOT_DIR, "convention.md"),
+    "utf8"
+  );
+
+  const date = new Date().toISOString().split("T")[0];
+  let builtConvention = `
+${generateSourceCommitComment(sourceCommitSha)}
+${sourceConvention.replace(VERSION_REGEX, version).replace(DATE_REGEX, date)}
+`.trim();
+
+  await fs.outputFile(path.join(OUTPUT_DIR, "README.md"), builtConvention);
+  await fs.copy(
+    path.join(ROOT_DIR, "LICENSE"),
+    path.join(OUTPUT_DIR, "LICENSE")
+  );
+}
+
+async function deployToGit(
+  version,
+  targetBranch,
+  repoSlug,
+  username,
+  password
+) {
+  const { stdout } = await runScript(`
+    cd ${OUTPUT_DIR}
+    git init
+    git config --global user.name "Travis CI"
+    git config --global user.email "travis@travis-ci.org"
+    git remote add origin https://{${username}:${password}@github.com/${repoSlug}
+    git add .
+    git commit -m "Release v${version}"
+    git push -f origin master:${targetBranch}
+  `);
```

Comment:
I agree with @davidgraeff. You are effectively replacing the complete master branch with a new one-commit-(...I don't even know what to call this). This is indeed very bad. Amending a public commit should be avoided, rewriting history should be a last resort, replacing the complete history is just ....no.

What is the purpose of the master branch? Why didn't you stay with master-only and the version tags system I proposed in #46. I am not saying that this was the perfect arrangement but why all the overhead and a script that will be hard to maintain and troubleshoot?